src/main/java/org/nixos/idea/psi/NixStringManipulator.kt

```diff
@@ -0,0 +1,37 @@
+package org.nixos.idea.psi
+
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.AbstractElementManipulator
+import com.intellij.refactoring.suggested.startOffset
+import org.nixos.idea.psi.impl.AbstractNixString
+import org.nixos.idea.util.NixIndStringUtil
+import org.nixos.idea.util.NixStringUtil
+
+class NixStringManipulator : AbstractElementManipulator<NixString>() {
```

Comment:
I do not know no, again I followed the example

Your guess is as good as mine! 

The name of `handleContentChange()` makes me suspect this might be called when you chage the text inside the editor fragment of the **guest** language (and this function changes the corresponding string AST node in the **host** language)